File: bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/util/CollectionUtil.java
package dev.robocode.tankroyale.botapi.util;

import java.util.*;

/**
 * Collection utility class.
 */
public final class CollectionUtil {

    // Hides constructor
    private CollectionUtil() {
    }

    /**
     * Creates a mutable list that is a copy of another list.
     *
     * @param list is the list to copy, where {@code null} results in returning an empty mutable list.
     * @return a mutable list that is a copy of the input list.
     */
    public static <T> List<T> toMutableList(List<T> list) {
        return list == null ? new ArrayList<>() : new ArrayList<>(list);
    }

    /**
     * Creates a mutable set that copies all items from a collection, but is removing duplicates.
     *
     * @param collection is the collection to copy, where {@code null} results in returning an empty mutable set.
     * @return a mutable set that is a copy of the input collection.
     */
    public static <T> Set<T> toMutableSet(Collection<T> collection) {
        return collection == null ? new HashSet<>() : new HashSet<>(collection);
    }
}

File: bot-api/dotnet/api/src/util/CollectionUtil.cs
using System.Collections.Generic;

namespace Robocode.TankRoyale.BotApi.Util;

/// <summary>
/// Collection utility class.
/// </summary>
static class CollectionUtil
{
    /// <summary>
    /// Creates a mutable list that is a copy of another list.
    /// </summary>
    /// <param name="list">The list to copy, where <c>null</c> results in returning an empty mutable list.</param>
    /// <returns>A mutable list that is a copy of the input list.</returns>
    internal static IList<T> ToMutableList<T>(IEnumerable<T> list) {
        return list == null ? new List<T>() : new List<T>(list);
    }

    /// <summary>
    /// Creates a mutable set that copies all items from a collection, but is removing duplicates.
    /// </summary>
    /// <param name="collection">The collection to copy, where <c>null</c> results in returning an empty mutable set.
    /// </param>
    /// <returns>A mutable set that is a copy of the input collection.</returns>
    internal static ISet<T> ToMutableSet<T>(ISet<T> collection) {
        return collection == null ? new HashSet<T>() : new HashSet<T>(collection);
    } 
}